sympy/core/tests/test_expr.py

```diff
@@ -1764,3 +1765,7 @@ def test_issue_10755():
     x = symbols('x')
     raises(TypeError, lambda: int(log(x)))
     raises(TypeError, lambda: log(x).round(2))
+
+def test_issue_11877():
+    x = symbols('x')
+    assert (long(integrate(log(0.5-x), (x, 0, 0.5))) == long(-0.846573590279973))
```

Comment:
This test probably shows that there is no imaginary part, but otherwise it does not say much:
```
In [3]: long(-0.846573590279973)
Out[3]: 0
```
It is usually best to use rational numbers in tests, since floating point comparisons may be problematic. In this case, `integrate(log(S(1)/2 - x), (x, 0, S(1)/2))` could be used.